# Example 1-1: Determining the Semimajor Axis from the Period
### _Fundamentals of Astrodynamics and Applications_, 5th Ed., 2022, p. 31-32

This notebook demonstrates finding the semimajor axis from the orbital period, assuming a geosynchronous orbit.

GIVEN:&nbsp;A geosynchronous orbit<br>
FIND:&nbsp;Semimajor axis, _a_

## Install and Import Libraries
---

First, install `valladopy` if it doesn't already exist in your environment:

In [1]:
!pip install valladopy

Then import `valladopy` to use its `constants` module:

In [2]:
from valladopy import constants as const

## Solution
---

The orbital period $\mathcal{P}$ is defined from Eq. 1-26:

$$
\mathcal{P} = 2\pi \sqrt{\frac{a^3}{\mu}}
$$

Rearranging:

$$
a = \left( \mu \left( \frac{\mathcal{P}}{2\pi} \right)^2 \right)^{1/3}
$$
<br>
where $\mathcal{P}$ is equal to 24 sidereal hours ($86,\!400 \div 1.002737909350795 = 86,\!164.090518$ sec), because the Earth completes one revolution through $2\pi$ radians in this amount of time. Be careful not to use 24 solar hours, which is a little *more* than one complete revolution.

In [3]:
a = (const.MU * (86164.090518 / const.TWOPI) ** 2) ** (1 / 3)

print(f'Semimajor axis:\t\t{a:.4f}\tkm')
print(f'\t\t\t{a/const.RE:.4f}\t\tEarth radii')

Semimajor axis:		42164.1696	km
			6.6107		Earth radii


A semimajor axis of ~42,164 km (or ~6.6 Earth radii) is consistent with a geostationary orbit.